GitHub_LearningTests/Feature/StartUp/Application/StartUpControllerTests.swift

```diff
@@ -0,0 +1,226 @@
+        setUpTests(
+            loadOnboardingFinishedFacade: facade,
+            credentialKeychainController: credentialKeychainController,
+            loginLauncherController: loginLauncherController
+        )
+        
+        // when
+        sut.startUp()
+        
+        //Then
+        XCTAssert(credentialKeychainController.isLoaded == false)
+        XCTAssertNotNil(loginLauncherController.launched)
+        XCTAssert(loginLauncherController.launched == true)
+        XCTAssert(loginLauncherController.point == LoginLaunchPoint.start)
+    }
+    
+    private class LoadOnboardingFinishedFacadeDummy: LoadOnboardingFinishedFacade {
+        func load(_ request: LoadOnboardingFinishedRequest, completion: @escaping ((Result<LoadOnboardingFinishedResponse>) -> Void)) {}
+    }
+    
+    private class LoadOnboardingFinishedFacadeSuccessTrueStub: LoadOnboardingFinishedFacade {
+        func load(_ request: LoadOnboardingFinishedRequest, completion: @escaping ((Result<LoadOnboardingFinishedResponse>) -> Void)) {
+            completion(.success(response: LoadOnboardingFinishedResponse(isFinished: true)))
+        }
+    }
+    
+    private class LoadOnboardingFinishedFacadeSuccessFalseStub: LoadOnboardingFinishedFacade {
+        func load(_ request: LoadOnboardingFinishedRequest, completion: @escaping ((Result<LoadOnboardingFinishedResponse>) -> Void)) {
+            completion(.success(response: LoadOnboardingFinishedResponse(isFinished: false)))
+        }
+    }
+    
+    private class LoadOnboardingFinishedFacadeFailureStub: LoadOnboardingFinishedFacade {
+        func load(_ request: LoadOnboardingFinishedRequest, completion: @escaping ((Result<LoadOnboardingFinishedResponse>) -> Void)) {
+            completion(.failure(error: LocalStorageError.loadFailed("Load Failed")))
+        }
+    }
+    
+    private class CredentialKeychainControllerDummy: TestController, CredentialKeychainController {
+        var isLoaded: Bool?
+    
+        func storeCredentials(_ credintals: String) {}
+        func deleteCredentials() {}
+        
+        func loadCredentials(_ completion: @escaping (Result<LoadCredentialsKeychainResponse>) -> Void) {
+            isLoaded = true
+        }
+    }
+    
+    private class CredentialKeychainControllerSuccessStub: TestController, CredentialKeychainController {
+        var isLoaded: Bool?
+        var loadedCredentials: String?
+    
+        func storeCredentials(_ credintals: String) {}
+        func deleteCredentials() {}
+        
+        func loadCredentials(_ completion: @escaping (Result<LoadCredentialsKeychainResponse>) -> Void) {
+            isLoaded = true
+            let credentials = "Success"
+            self.loadedCredentials = credentials
+            completion(.success(response: LoadCredentialsKeychainResponse(credentials: credentials)))
+        }
+    }
+    
+    private class CredentialKeychainControllerFailureStub: TestController, CredentialKeychainController {
```

Comment:
Ty bys tady nakonec vubec nepotreboval ty Success a Failure Stuby, stacil by ti jeden, lae v jeho initu bys predal to co ma vracet. Pak bys na to mel referenci i v tele testu a mohl to v Then casti porovnavat.